drivers/shared/resolvconf/mount.go

```diff
@@ -0,0 +1,80 @@
+package resolvconf
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+
+	dresolvconf "github.com/docker/libnetwork/resolvconf"
+	"github.com/docker/libnetwork/types"
+	"github.com/hashicorp/nomad/plugins/drivers"
+)
+
+func GenerateDNSMount(taskDir string, conf *drivers.DNSConfig) (*drivers.MountConfig, error) {
+	var nSearches, nServers, nOptions int
+	path := filepath.Join(taskDir, "resolv.conf")
+	mount := &drivers.MountConfig{
+		TaskPath:        "/etc/resolv.conf",
+		HostPath:        path,
```

Comment:
It looks like we're only using this function for `exec` and `java` task drivers. When we pass the `TaskDir` into this function to create the `HostPath`, is that path writable to the task? Mounting it from there read-only to `/etc/resolv.conf` makes it seem like it shouldn't be. It doesn't seem like a security issue because it's a copy of the host's `/etc/resolv.conf` and not a mount to it. Are we doing this bind-mount mostly for consistency with how we create mounts?

Reply:
Yeah the idea being that if Nomad is going to manage the dns configuration (resolv.conf) then the task shouldn't be able to write to it. I couldn't find a great place to write the resolv.conf as we don't really have a per alloc directory thats not actually accessible from the alloc. I figured mounting as ro would give the "Nomad manages this file" UX even if we change where its mounted from in the future. 

I did try and keep the concept of just passing a bind mount consistent as that seems the most compatible way other drivers could consume this.

Reply:
👍 That all sounds like solid reasoning.